lib/posix/unique_file_descriptor_test.cc

```diff
@@ -0,0 +1,10 @@
+#include "lib/posix/unique_file_descriptor.h"
+#include "gtest/gtest.h"
+
+using namespace posix;
+
+TEST(UniqueFileDescriptorTest, DefaultConstruct) { UniqueFileDescriptor fd; }
+
+TEST(UniqueFileDescriptorTest, Invalid) {
+  UniqueFileDescriptor fd(kInvalidFileDescriptor);
+}
```

Comment:
What are these tests asserting?

Reply:
I didn't answer the core of your question. These tests are mainly checking that the code compiles since this is not guaranteed until the first instantiation of a templated type.